src/store/rocket.js

```diff
@@ -0,0 +1,74 @@
+import _ from 'lodash'
+import { WebClient } from '@slack/web-api'
+
+const state = () => {
+    return {employeeHeadshots: null, 
+      employeesMissingFromSite: null
+    }
+  };
+  
+  const mutations = {
+    setEmployeeHeadshots(state, employeeHeadshots) {
+        state.employeeHeadshots = employeeHeadshots;
+      },
+    setEmployeesMissingFromSite(state, payload) {
+        const uniqueEmployees = _.differenceBy(payload, state.employeeHeadshots, 'name')
+        state.employeesMissingFromSite = uniqueEmployees;
+      }
+  };
+  
+  const actions = {
+    async getContentfulEmployees ({ commit }) {
+        try{
+            const employees = await this.$http.$get(this.$config.contentfulFullUrl)
+            const itemsFields = ['fields.name', 'fields.position', 'fields.profilePic.sys.id','sys.id']
+            const itemsFilteredEmployees = _.map(employees.items, e => _.pick(e, itemsFields))
+            const includesFields = ['fields.file.fileName', 'fields.file.url', 'sys.id']
+            const includesFilteredEmployees = _.map(employees.includes.Asset, e => _.pick(e, includesFields))
+            const mergedFilteredEmployees = _.values(_.merge(_.keyBy(includesFilteredEmployees,'sys.id'), _.keyBy(itemsFilteredEmployees,'fields.profilePic.sys.id' ) ))
+            const formattedFinalFilteredEmployees = _.map(mergedFilteredEmployees, item => {
+            return {
+                        id:item.sys.id, 
+                        name:item.fields.name, 
+                        position:item.fields.position, 
+                        imageId:item.fields.profilePic.sys.id,
+                        imageName:item.fields.file.fileName, 
+                        imageUrl:item.fields.file.url
+                    }
+            }
+            )
+            commit('setEmployeeHeadshots', formattedFinalFilteredEmployees)
+        } catch(error) {
+            console.error(error.message)
+        }
+        
+  }, 
+    async getSlackEmployees ({ commit }) {
+      try{
+          const web = new WebClient(this.$config.slackBearerToken)
+          const slackEmployees = await web.users.list({team_id:'T03TAQHEU'})
+          console.log("not filtered by is_bot", slackEmployees)
+          const notBotEmployees = _.filter(slackEmployees.members, e => !e.is_bot)
+          console.log("filtered by is_bot",notBotEmployees)
+          const formattedNotBotEmployees = _.map(notBotEmployees , item => {
```

Comment:
First, a note on the word "collection." I tend to use it as "an array of objects where every object has the same structure." Lodash uses that word to mean a standard array _or_ an array of objects.

This filter and map could be combined into one step. Since you're mapping over the collection, which means you're going to iterate over every item of the collection, you can do your filtering and assignment in the same step. That way you're not looping over the same data twice. The map step can check for that `is_bot` flag... but... a twist!

Since mapping makes you return something for each iteration of the collection, that means the length of the input collection will be the length of the output collection. But, because of `is_bot`, we actually don't want that. If we stuck with map, we'd have to return `undefined` or `false` and then use something like `_.compact` to remove those falsey values.

But... there's reduce. Reduce is rad AF and so many people overlook it. But map/reduce is what made "big data" a whole thing for a few years so it's worth knowing. Suffice to say, knowing map and reduce well will take you very far for just about any structure of data you'll come across.

With reduce, you're boiling down your data, and you can end up with whatever you want. An array, a number, an object, etc. When using reduce, the function that gets called on each loop gets a new first argument: the "total" or the thing you're building up as you go. If that's an array, you're building up array items as you go. If it's an object, you're building up object properties as you go. If it's a number, you're doing math as you go. That function for the reduce method _must_ return the new "total" every time. This is a common example, using our `is_bot` for inspiration (note that I've referred to our "total" here as `list`):
```
// note the third argument, the empty array: that's the default
// value reduce starts with, so on the first loop, `list === []`

_.reduce(collection, (list, user, index) => {
   if (user.is_bot) return list
   return [...list, user]
}, []}
```

That's along the lines of what you're looking for here. Reduce is fun to play with, but it definitely takes some practice and muscle memory before you'll reach for it right away.